support-models/src/main/scala/com/gu/support/encoding/JsonHelpers.scala

```diff
@@ -98,4 +98,11 @@ object JsonHelpers {
     def getField(key: String) = json.hcursor.downField(key).focus
   }
 
+  //Decodes an expected json string to T if the given partial function is defined for the string value
+  def decodeStringAndCollect[T](pf: PartialFunction[String,T]): Decoder[T] = Decoder.decodeString.emap { s =>
```

Comment:
is this what works in conjunction with the stuff in `ContributionTypes.scala` to encode/decode a sum type to JSON? (asking for my own Scala curiosity!)

Reply:
yep, it's just a convenient way to match on a json string, for case object decoding